src/mono/msbuild/apple/build/AppleBuild.props

```diff
@@ -26,6 +26,8 @@
         _InitializeCommonProperties;
         _BeforeAppleBuild;
         _AppleResolveReferences;
+        _ScanAssembliesDecideLightweightMarshaler;
+        _ProcessRuntimeComponents;
```

Comment:
NIT: I think `_ProcessRuntimeComponents` target could be refactored a bit, especially as it has a single condition `_IsLibraryMode` on an item group at https://github.com/dotnet/runtime/pull/88903/files#diff-e351a598f31c0a001f31622d9babc297dbcbe5827eabf55ab92700e1839e70f1R109 
 
Proposed changes:
1. Add a property depending on library mode builds
```xml
  <PropertyGroup Condition="'$(_IsLibraryMode)' == 'true'">
   <_ProcessRuntimeComponentsForLibraryMode>_ProcessRuntimeComponentsForLibraryMode</_ProcessRuntimeComponentsForLibraryMode>
  </PropertyGroup>
``` 
2. Adjust the `AppleBuildDependsOn` targets
```xml
<AppleBuildDependsOn>
        _InitializeCommonProperties;
        _BeforeAppleBuild;
        _AppleResolveReferences;
        _ScanAssembliesDecideLightweightMarshaler;
        $(_ProcessRuntimeComponentsForLibraryMode);
        _AppleAotCompile;
        ...
``` 
3. Rename the `_ProcessRuntimeComponents` target into `_ProcessRuntimeComponentsForLibraryMode`
4. Remove the condition on the item group on line: https://github.com/dotnet/runtime/pull/88903/files#diff-e351a598f31c0a001f31622d9babc297dbcbe5827eabf55ab92700e1839e70f1R109

Reply:
Updated.